ui/admin/app/components/form/worker/index.hbs

```diff
@@ -0,0 +1,87 @@
+<Rose::Form
+  @onSubmit={{@submit}}
+  @cancel={{@cancel}}
+  @disabled={{@model.isSaving}}
+  @error={{@model.errors.base}}
+  @showEditToggle={{if @model.isNew false true}}
+  as |form|
+>
+
+  {{#unless form.isEditable}}
+    <form.input
+      @name='type'
+      @value={{@model.type}}
+      @label={{t 'form.type.label'}}
+      @helperText={{t 'form.worker_type.help'}}
+      readonly={{true}}
+    />
+  {{/unless}}
+
+  <form.input
+    @name='name'
+    @type='name'
+    @value={{@model.name}}
+    @label={{t 'form.name.label'}}
+    @helperText={{t 'form.name.help'}}
+    @error={{@model.errors.name}}
+    readonly={{false}}
+    as |field|
+  >
+    {{#if @model.errors.name}}
+      <field.errors as |errors|>
+        {{#each @model.errors.name as |error|}}
+          <errors.message>{{error.message}}</errors.message>
+        {{/each}}
+      </field.errors>
+    {{/if}}
+  </form.input>
+
+  <form.textarea
+    @name='description'
+    @type='description'
+    @value={{@model.description}}
+    @label={{t 'form.description.label'}}
+    @helperText={{t 'form.description.help'}}
+    @error={{@model.errors.description}}
+    disabled={{@model.isSaving}}
+    readonly={{false}}
+    as |field|
+  >
+    {{#if @model.errors.description}}
+      <field.errors as |errors|>
+        {{#each @model.errors.description as |error|}}
+          <errors.message>{{error.message}}</errors.message>
+        {{/each}}
+      </field.errors>
+    {{/if}}
+  </form.textarea>
+
+  {{#unless form.isEditable}}
+    <form.input
+      @name='address'
+      @type='url'
+      @value={{@model.address}}
+      @label={{t 'form.address.label'}}
+      @helperText={{t 'form.worker_address.help'}}
+      readonly={{true}}
+    />
+
+    <form.input
+      @name='last seen'
+      @type='datetime-local'
+      @value={{@model.last_status_time}}
+      @label={{t 'form.worker_last_seen.label'}}
+      @helperText={{t 'form.worker_last_seen.help'}}
+      readonly={{true}}
+    />
+  {{/unless}}
+
+  {{#if (can 'save model' @model)}}
```

Comment:
Technically it doesn't change anything but we should use the `worker` model instead of base `model`

Reply:
@ZedLi not following what you mean here.  Isn't this just away for authorizing what users have the correct role to edit the form?

Reply:
Right, but we also made a custom `worker` ability, it'll still check the `canSave` on the base model but we should use the more specific `worker` one since it exists.